### PR TITLE
Add some privatekey password options

### PR DESCRIPTION
**Pull request by zhangjunjie6b**
Provide password configuration of privatekey

### CODE REVIEW

**Comment by zhangjunjie6b** [comment]
I'll change it again

File: README.md [hunk 1]
```diff
@@ -679,6 +680,7 @@ ssh:
       username: ubuntu  # SSH Login username
       password: xxxxx   # SSH Login password
       key: /path/to/private.key # SSH login private file
+      private_password: xxxxxxx  # PrivateKey password
```
Comment (haoel) [inline README.md:683]: I think use the name `passphrases` might be better.

**Review by haoel — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (zhangjunjie6b) [reply README.md]: me too

Comment (zhangjunjie6b) [reply README.md]: > I think use the name `passphrases` might be better.

good idea

File: README.md [hunk 2]
```diff
@@ -104,7 +104,7 @@ On application startup, the configured probes are scheduled for their initial fi
         host: ubuntu@172.10.1.1:22
         password: xxxxxxx
         key: /Users/user/.ssh/id_rsa
-        private_password: xxxxxxx
+        passphrases: xxxxxxx
```
Comment (proditis) [inline README.md:107]:
```suggestion
        passphrase: xxxxxxx
```
Since we're referring to one make the word singular

File: README.md [hunk 3]
```diff
@@ -680,7 +680,7 @@ ssh:
       username: ubuntu  # SSH Login username
       password: xxxxx   # SSH Login password
       key: /path/to/private.key # SSH login private file
-      private_password: xxxxxxx  # PrivateKey password
+      passphrases: xxxxxxx  # PrivateKey password
```
Comment (proditis) [inline README.md:683]:
```suggestion
      passphrase: xxxxxxx  # PrivateKey password
```
same

File: probe/ssh/endpoint.go [hunk 4]
```diff
@@ -29,12 +29,12 @@ import (
...
-	Host            string      `yaml:"host"`
-	User            string      `yaml:"username"`
-	Password        string      `yaml:"password"`
-	PrivatePassword string      `yaml:"private_password"`
-	client          *ssh.Client `yaml:"-"`
+	PrivateKey  string      `yaml:"key"`
+	Host        string      `yaml:"host"`
+	User        string      `yaml:"username"`
+	Password    string      `yaml:"password"`
+	Passphrases string      `yaml:"passphrases"`
```
Comment (proditis) [inline probe/ssh/endpoint.go:36]:
```suggestion
	Passphrase string      `yaml:"passphrase"`
```

File: probe/ssh/endpoint.go [hunk 5]
```diff
@@ -82,8 +82,8 @@ func (e *Endpoint) SSHConfig(kind, name string, timeout time.Duration) (*ssh.Cli
 		// Create the Signer for this private key.
 		var signer ssh.Signer
 
-		if len(e.PrivatePassword) > 0 {
-			signer, err = ssh.ParsePrivateKeyWithPassphrase(key, []byte(e.PrivatePassword))
+		if len(e.Passphrases) > 0 {
```
Comment (proditis) [inline probe/ssh/endpoint.go:85]:
```suggestion
		if len(e.Passphrase) > 0 {
```

File: probe/ssh/endpoint.go [hunk 6]
```diff
@@ -82,8 +82,8 @@ func (e *Endpoint) SSHConfig(kind, name string, timeout time.Duration) (*ssh.Cli
...
+			signer, err = ssh.ParsePrivateKeyWithPassphrase(key, []byte(e.Passphrases))
```
Comment (proditis) [inline probe/ssh/endpoint.go:86]:
```suggestion
			signer, err = ssh.ParsePrivateKeyWithPassphrase(key, []byte(e.Passphrase))
```

**Review by proditis — COMMENTED** [review]
LGTM

File: probe/ssh/endpoint.go (hunk 6 above)
Comment (zhangjunjie6b) [reply probe/ssh/endpoint.go]: sorry， my english is not good  :(

Comment (haoel) [reply probe/ssh/endpoint.go]: no need to say sorry, I guide you to the improper word. ;-)

**Comment by haoel** [comment]
One more thing, can you add the unit test case for this new feature? Thanks so much! @zhangjunjie6b

File: probe/ssh/endpoint.go (hunk 6 above)
Comment (zhangjunjie6b) [reply probe/ssh/endpoint.go]: it's better than `private_password` :-)

**Review by haoel — APPROVED** [review]
Thanks for this contribution!

**Pull request merged**